code/modules/mob/living/simple_animal/guardian/guardian.dm

```diff
@@ -593,15 +593,22 @@
 	if(candidates.len)
 		theghost = pick(candidates)
 		var/mob/living/simple_animal/hostile/guardian/G = spawn_guardian(user, theghost.key)
-		var/time_limit = world.time + 300
-		var/guardianNewName = stripped_input(G.summoner, "You are the user of [G.name]. Would you like to name your guardian something else?", "Name Guardian", G.name, MAX_NAME_LEN)
-		if(world.time < time_limit)
-			if(length(guardianNewName) > 0)
+		//Give the stand user 3 chances to rename their stand
+		for(var/i = 2, i >= 0,i--)
+			var/guardianNewName = stripped_input(user, "You are the user of [G.name]. Would you like to name your guardian something else?", "Name Guardian", G.name, MAX_NAME_LEN)
```

Comment:
I can re-implement that feature pretty easily.
I'll refactor the attempt thing.
Attempt thing works properly.

That's an invalid name! You have 2 more attempts.
That's an invalid name! You have 1 more attempt.
Sorry, you've ran out of attempts! Looks like you're stuck with Guardian Spirit Cyan!

The first line appears after you fail the first input
Second line appears appears you fail the second
Third appears after the third